src/utils/cosmosdb_model_versioned_ttl.ts

```diff
@@ -0,0 +1,234 @@
+    protected readonly modelIdKey: ModelIdKey,
+    protected readonly partitionKey?: PartitionKey
+  ) {
+    super(
+      container,
+      newVersionedItemT,
+      retrievedItemT,
+      modelIdKey,
+      partitionKey
+    );
+  }
+
+  /**
+   * Using transactionalBatch updates the ttl field for all the versions
+   * of the document identified by the searchKey.
+   *
+   * @param searchKey
+   * @param ttl
+   * @returns Either a Cosmos Error or the number of updated elements
+   */
+  public updateTTLForAllVersions(
+    searchKey: DocumentSearchKey<TR, ModelIdKey, PartitionKey>,
+    ttl: RetrievedVersionedModelTTL["ttl"]
+  ): TE.TaskEither<CosmosErrors, number> {
+    const partitionKey = ((searchKey.length === 1
+      ? searchKey[0]
+      : searchKey[1]) as unknown) as TN[PartitionKey];
+
+    return pipe(
+      this.findAllVersionsBySearchKey(searchKey),
+      TE.map(RA.rights),
+      TE.map(
+        RA.map(doc => ({
+          id: doc.id,
+          operationType: "Patch" as const,
+          resourceBody: {
+            operations: [
+              {
+                // `add` will create or update the path with the passed value
+                op: PatchOperationType.add,
+                path: `/ttl`,
+                value: ttl
+              }
+            ]
+          }
+        }))
+      ),
+      TE.map(RA.chunksOf(100)),
+      TE.map(
+        RA.map(chunk =>
+          pipe(
+            chunk,
+            RA.toArray,
+            operations => this.batch(operations, partitionKey),
+            TE.chainW(
+              TE.fromPredicate(
+                response => response.code === 200,
+                _ => {
+                  const firstChunkId = chunk[0]?.id ?? -1;
+                  const lastChunkId = chunk[chunk.length - 1]?.id ?? -1;
+
+                  return CosmosErrorResponse({
+                    message: `Error updating ttl for ${searchKey} - chunk from ${firstChunkId} to ${lastChunkId}`,
+                    name: `Error updating ttl`
+                  });
+                }
+              )
+            )
+          )
+        )
+      ),
+      TE.chainW(TE.sequenceSeqArray),
+      TE.chainW(responses =>
+        pipe(
+          responses,
+          RA.filter(response => response.code !== 200),
+          TE.fromPredicate(
+            errors => errors.length === 0,
+            _ =>
+              // Responses have status code 200, since check has been make within previous step
+              // so, this should never happen
+              CosmosErrorResponse({
+                message: `Error updating ttl for ${searchKey}. WARNING: This error should never happen!`,
+                name: `Error updating ttl`
+              })
+          ),
+          TE.chainW(() =>
+            pipe(
+              responses,
+              RA.reduce(0, (v, r) => v + (r.result?.length ?? 0)),
+              TE.right
+            )
+          )
```

Comment:
This as been checked before.
```suggestion
        pipe(
          responses,
          RA.reduce(0, (v, r) => v + (r.result?.length ?? 0)),
          TE.right
        )
```

Reply:
You're right, I just put a error for that should never be raised ("WARNING: This error should never happen!") just for check.
Do you think it's safe to remove it?